controller/volume_controller.go

```diff
@@ -2358,14 +2394,197 @@ func (c *VolumeController) getReplicaCountForAutoBalanceBestEffort(v *longhorn.V
 	}
 
 	if mostExtraRCount == 0 || mostExtraRCount == leastExtraRCount {
-		return 0, nil, []string{}
+		mostExtraRList = nil
+		leastExtraROwners = []string{}
+	} else {
+		adjustCount = mostExtraRCount - leastExtraRCount - 1
+		log.Infof("Found %v replicas from %v to balance to one of node in %v", adjustCount, mostExtraRList, leastExtraROwners)
+	}
+
+	if adjustCount == 0 {
+		replicasUnderDiskPressure, err := c.getReplicasUnderDiskPressure()
```

Comment:
if `replica-soft-anti-affinity` is `disabled`, is `replicasUnderDiskPressure` be an empty map?

Reply:
I am also fine with removing the dependency on `replica-soft-anti-affinity` setting.

https://github.com/longhorn/longhorn/pull/7576#issuecomment-2190479295